applications/serial_lte_modem/src/main.c

```diff
@@ -53,6 +53,16 @@ static void indicate_wk(struct k_work *work);
 
 BUILD_ASSERT(CONFIG_SLM_WAKEUP_PIN >= 0, "Wake up pin not configured");
 
+NRF_MODEM_LIB_ON_INIT(serial_lte_modem_init_hook, on_modem_lib_init, NULL);
+
+/* Initialized to value different than success (0) */
+static int modem_lib_init_result = -1;
+
+static void on_modem_lib_init(int ret, void *ctx)
+{
```

Comment:
Will this work for the scenario of modem firmware FOTA?
There is an query of intermediate modem init result so as to do a reset to complete FOTA.

Reply:
@junqingzou Functionality should be equivalent to the previous implementation

Reply:
Thanks, will re-test modem FOTA after this PR is merged.